src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -94,6 +100,37 @@ func (r *StatefulSetResource) runUpdate(
 	return nil
 }
 
+func (r *StatefulSetResource) isClusterHealthy(ctx context.Context) error {
+	if !featuregates.ClusterHealth(r.pandaCluster.Status.Version) {
+		r.logger.V(debugLogLevel).Info("Cluster health endpoint is not available", "version", r.pandaCluster.Spec.Version)
+		return nil
+	}
+
+	adminAPIClient, err := r.getAdminAPIClient(ctx)
```

Comment:
nit: not needed for this PR but admin api client is used everywhere and each resource have it's own implementation (console, cluster, sts). i think we should put this to util or some other way so we DRY

Reply:
Yes, but there is problem with commonality. I'm happy to be wrong, but I don't see it yet.